packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -35,11 +46,34 @@ class WindowsVersionValidator extends DoctorValidator {
     // Use the string split method to extract the major version
     // and check against the [kUnsupportedVersions] list
     final ValidationType windowsVersionStatus;
-    final String statusInfo;
+    final List<ValidationMessage> messages = <ValidationMessage>[];
+    String statusInfo;
     if (matches.length == 1 &&
         !kUnsupportedVersions.contains(matches.elementAt(0).group(1))) {
-      windowsVersionStatus = ValidationType.success;
       statusInfo = 'Installed version of Windows is version 10 or higher';
+
+      // Check if the Topaz OFD security module is running, and warn the user if it is.
+      // See https://github.com/flutter/flutter/issues/121366
+      final ProcessResult getProcessesResult = await _processLister.getProcessesWithPath();
+      if (getProcessesResult.exitCode != 0) {
+        windowsVersionStatus = ValidationType.partial;
+        statusInfo = 'Failed to execute Get-Process';
```

Comment:
One more change... can we move all of the logic into a private method for this validator instead actually and have it return a bool, or some other indicator that we can use conditionals against to edit the `statusInfo` and `messages` list?

Doing that will make it easier if we have to add more windows checks in the future and also makes it easier to understand the logic.

You can invoke this method once we have confirmed that the windows version is valid

Reply:
Would we actually want a `bool` out of that? It seems to me that we would also want the status info and/or validation messages list to depend on this logic, and they may have more than just 2 possible states.

Reply:
Yeah agreed, the `bool` was just an example of what can be returned. You can return a pair of values, one for status and another for messages that get joined together.

This is to just make it so that we can maintain this validator easily in the future and possibly add more if needed

Reply:
It sounds like this private method would then just essentially return a `ValidationResult`. What logic are you referring to moving? Just the version check, the processes check, or both? If both, would not `validate` just turn into a wrapper around calling this private method? I am not sure I understand what change you are proposing.

Reply:
Yeah, sorry if it's not coming off clear, the below is what I was thinking

```dart
    // Use the string split method to extract the major version
    // and check against the [kUnsupportedVersions] list
    ValidationType windowsVersionStatus;
    String statusInfo;
    final List<ValidationMessage> messages = <ValidationMessage>[];
    if (matches.length == 1 && !kUnsupportedVersions.contains(matches.elementAt(0).group(1))) {
      windowsVersionStatus = ValidationType.success;
      statusInfo = 'Installed version of Windows is version 10 or higher';
      messages.add(ValidationMessage(statusInfo));

      // List of windows specific checks which can be unrelated to versions but
      // could be causing users problems
      final List<ValidationResult> windowsSubResults = [
        _topazFinder(), // The private method with the logic needed
      ];
      for (final result in windowsSubResults) {
        if (result.type != ValidationType.success) {
          statusInfo = 'Problem detected with Windows installation';  // This should be more generic while we list the problem in the messages
          windowsVersionStatus = Validation.partial;
          messages.add(result.message);
        }
      }

    } else {
      windowsVersionStatus = ValidationType.missing;
      statusInfo = 'Unable to determine Windows version (command `ver` returned $commandResult)';
    }
```

This way if we need to add more checks like yours in the future it should be easy for us to add it to the list of checks

Also, for this to work, I walk back my comment about making the return variables `final`.